test/unit/modules/blocks/utils.js

```diff
@@ -13,268 +13,558 @@
+				loadBlocksData: sinonSandbox.stub(),
+				aggregateBlocksReward: sinonSandbox.stub().resolves(),
+			},
+		};
+
+		dbStub.blocks.loadBlocksData
+			.withArgs(sinonSandbox.match({ id: '13068833527549895884' }))
+			.resolves(viewRow_full_blocks_list)
+			.withArgs(sinonSandbox.match({ id: '1' }))
+			.resolves([]);
+
+		loggerStub = {
+			trace: sinonSandbox.spy(),
+			info: sinonSandbox.spy(),
+			error: sinonSandbox.spy(),
+		};
+
+		blockMock = {
+			dbRead: function(input) {
+				return { id: input.b_id, height: input.b_height };
+			},
+		};
+
+		transactionMock = {
+			dbRead: function(input) {
+				return { id: input.t_id, type: input.t_type };
+			},
+		};
+
+		accountMock = {
+			get: sinonSandbox.stub(),
+		};
+
+		accountMock.get
+			.withArgs(sinonSandbox.match({ address: 'ERRL' }))
+			.callsArgWith(1, 'Address error stub', null)
+			.withArgs(sinonSandbox.match({ address: '0L' }))
+			.callsArgWith(1, null, undefined)
+			.withArgs(sinonSandbox.match({ address: '1L' }))
+			.callsArgWith(1, null, { publicKey: '123abc' });
+
+		blocksUtilsModule = new BlocksUtils(
+			loggerStub,
+			accountMock,
+			blockMock,
+			transactionMock,
+			dbStub,
+			modulesLoader.scope.dbSequence,
+			modulesLoader.scope.genesisblock
+		);
 
-			describe('should assign logic', () => {
-				it('should assign block');
+		modulesStub = {
+			blocks: {
+				lastBlock: {
+					get: sinonSandbox
+						.stub()
+						.returns({ id: '9314232245035524467', height: 1 }),
+					set: sinonSandbox
+						.stub()
+						.returns({ id: '9314232245035524467', height: 1 }),
+				},
+				utils: {
+					readDbRows: blocksUtilsModule.readDbRows,
+				},
+			},
+		};
+
+		blocksUtilsModule.onBind(modulesStub);
+
+		__private = BlocksUtils.__get__('__private');
+		__private.loaded = false;
+
+		library = BlocksUtils.__get__('library');
+		modules = BlocksUtils.__get__('modules');
+	});
 
-				it('should assign transaction');
-			});
-		});
+	afterEach(() => {
+		sinonSandbox.reset();
+	});
 
-		it('should set self to this');
+	describe('constructor', () => {
+		it('should assign params to library', () => {
+			expect(library.logger).to.eql(loggerStub);
+			expect(library.db).to.eql(dbStub);
+			expect(library.dbSequence).to.eql(modulesLoader.scope.dbSequence);
+			expect(library.logic.account).to.eql(accountMock);
+			expect(library.logic.block).to.eql(blockMock);
+			expect(library.logic.transaction).to.eql(transactionMock);
+		});
 
-		it(
-			'should call library.logger.trace with "Blocks->Utils: Submodule initialized."'
-		);
+		it('should call library.logger.trace with "Blocks->Utils: Submodule initialized."', () => {
+			expect(loggerStub.trace.args[0][0]).to.equal(
+				'Blocks->Utils: Submodule initialized.'
+			);
+		});
 
-		it('should return self');
+		it('should return self', () => {
+			expect(blocksUtilsModule).to.be.an('object');
+			expect(blocksUtilsModule.readDbRows).to.be.a('function');
+		});
 	});
 
 	describe('readDbRows', () => {
```

Comment:
We should test here also:
- if library.logic.block.dbRead was called with proper params and how many times
- library.logic.transaction.dbRead was called with proper params and how many times
- test path with more than one block passed
- test path with empty array passed

Reply:
Ok thanks.

Reply:
Done.